python/docs/source/development/debugging.rst

```diff
@@ -0,0 +1,187 @@
+..  Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+..    http://www.apache.org/licenses/LICENSE-2.0
+
+..  Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+=================
+Debugging PySpark
+=================
+
+PySpark uses Spark as an engine. If a PySpark application does not require interaction
+between Python workers and JVMs, Python workers are not launched. They are lazily launched only when
+Python native functions or data have to be handled, for example, when you execute pandas UDFs or
+PySpark RDD APIs.
+
+This page describes how to debug such Python applications and workers instead of focusing on debugging with JVM.
+Profiling and debugging JVM is described at `Useful Developer Tools <https://spark.apache.org/developer-tools.html>`_.
+
+
+Remote Debugging (PyCharm)
```

Comment:
Can we explicitly state that the remote debugger is available only PyCharm Professional (I know, it is clearly explained in PyCharm docs, but not everyone will get there)? And maybe link to [pydev remote debugger docs](https://www.pydev.org/manual_adv_remote_debugger.html) as an alternative.

Reply:
ahhhh .. I didn't know it's only for PyCharm Professional ..

Reply:
> And maybe link to pydev remote debugger docs as an alternative.

This one, I think we'll leave it as a future work. It'd be nice if somebody document this as well.